lib/elixir/src/elixir_erl.erl

```diff
@@ -559,6 +581,38 @@ get_type_docs(Set, Types) ->
      {_, Line, Doc} <- ets:lookup(Set, {Kind, Tuple})]
   ).
 
+translate_doc({{Name, Arity}, Line, Kind, Signature, Doc}) ->
+  {{translate_doc_kind(Kind), Name, Arity},
+   erl_anno:new(Line),
+   [signature_to_binary(Name, Signature)],
+   doc_value(Doc),
+   #{signature => Signature}
+  };
+translate_doc({{Name, Arity}, Line, Kind, Doc}) ->
+  {{translate_doc_kind(Kind), Name, Arity},
+   erl_anno:new(Line),
+   [],
+   doc_value(Doc),
+   #{}
+  }.
+
+translate_doc_kind(def)           -> function;
+translate_doc_kind(callback)      -> callback;
+translate_doc_kind(type)          -> type;
+translate_doc_kind(opaque)        -> opaque;
+translate_doc_kind(defmacro)      -> macro;
+translate_doc_kind(macrocallback) -> macrocallback.
+
+signature_to_binary(Name, Signature) ->
+  BinaryName = atom_to_binary(Name, utf8),
+  Args = lists:map(fun
+    FormatArg({'\\\\', _, [Left, Right]}) ->
+      [FormatArg(Left), " \\\\ ", 'Elixir.Macro':to_string(Right)];
+    FormatArg({Var, _, _}) ->
+      atom_to_binary(Var, utf8)
+  end, Signature),
+  iolist_to_binary([BinaryName, "(", lists:join(", ", Args), ")"]).
```

Comment:
Here we're also generating the function signatures (similarly to how IEx does it). This might make it unnecessary to include the original signature list (the simplified AST), but I added it in the metadata map for now. I'll see if it's necessary to keep it while tackling the last todo of migrating all functionality. Or we could decide to keep it in the metadata map just in case. I was also thinking whether to generate similar signatures for other kinds (callbacks, etc.) I think that would require touching an earlier compilation step that adds the necessary information to the ETS tables, but I haven't dug into it yet.

Reply:
We should trust the signature in the docs. :) So we can remove the signature from the metadata.